src/Renci.SshNet/Security/Cryptography/Ciphers/AesCipher.cs

```diff
@@ -1,559 +1,16 @@
 ﻿using System;
-using System.Globalization;
-
-using Renci.SshNet.Common;
+using System.Security.Cryptography;
 
 namespace Renci.SshNet.Security.Cryptography.Ciphers
 {
     /// <summary>
     /// AES cipher implementation.
     /// </summary>
-    public sealed class AesCipher : BlockCipher
+    public sealed class AesCipher : BlockCipher, IDisposable
```

Comment:
This `IDisposable` is problematic. `AesCipher` is created in some places, like here:

https://github.com/sshnet/SSH.NET/blob/2eec748b184e6ef3dfec93649c1ca1bdc27a627a/src/Renci.SshNet/ConnectionInfo.cs#L363C43-L378

but there is no Dispose call anywhere, so in the current state of this PR I cannot merge it.

Reply:
All of the `Key`, `DigitalSignature`, `HashAlgorithm` classes are `IDisposable` but we do not dispose them in `ConnectionInfo`.

I don't think this introduces any new problems. I can remove the `IDisposable` if you prefer. Either way the `IDisposable` members would not be disposed.

Reply:
They should be disposed in `Session`. The `ConnectionInfo` has only a function that create `Key`, `DigitalSignature`, `HashAlgorithm`

But it is not true that they are not used:
https://github.com/sshnet/SSH.NET/blob/2eec748b184e6ef3dfec93649c1ca1bdc27a627a/src/Renci.SshNet/Session.cs#L1427-L1438

I don't want to say that the current implementation is perfect, but we just introduced a new problem in this PR. I didn't check how important it is.

Reply:
You're right, I will add a check for `IDisposable` in `Session`